storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -3897,15 +3905,37 @@ void xtrabackup_backup_func(void) {
   crc_init();
 
   xb_filters_init();
+  if (!validate_options("my", orig_argc, orig_argv)) {
+    exit(EXIT_FAILURE);
+  }
+  /* create extra LSN dir if it does not exist. */
+  if (xtrabackup_extra_lsndir &&
+      !my_stat(xtrabackup_extra_lsndir, &stat_info, MYF(0)) &&
+      (my_mkdir(xtrabackup_extra_lsndir, 0777, MYF(0)) < 0)) {
+    msg("xtrabackup: Error: cannot mkdir %d: %s\n", my_errno(),
+        xtrabackup_extra_lsndir);
+    exit(EXIT_FAILURE);
+  }
 
-  if (!xb_keyring_init_for_backup(mysql_connection)) {
-    msg("xtrabackup: Error: failed to init keyring plugin\n");
+  /* create target dir if not exist */
+  if (!my_stat(xtrabackup_target_dir, &stat_info, MYF(0)) &&
+      (my_mkdir(xtrabackup_target_dir, 0777, MYF(0)) < 0)) {
+    msg("xtrabackup: Error: cannot mkdir %d: %s\n", my_errno(),
+        xtrabackup_target_dir);
     exit(EXIT_FAILURE);
```

Comment:
any reason for moving them earlier?

Reply:
No needed. I reverted this change. It was required to write the xtrabackup_component_keyring_file.cnf at the start of the backup, but I have deferred the write operation to backup_finish function.